src/Symfony/Component/Form/FormError.php

```diff
@@ -187,4 +187,18 @@ public function unserialize($serialized)
     {
         list($this->message, $this->messageTemplate, $this->messageParameters, $this->messagePluralization, $this->cause) = unserialize($serialized);
     }
+
+    /**
+     * @see JsonSerializable::jsonSerialize()
+     */
+    public function jsonSerialize()
+    {
+        return array(
+            'message' => $this->message,
+            'messageTemplate' => $this->messageTemplate,
+            'messageParameters' => $this->messageParameters,
+            'messagePluralization' => $this->messagePluralization,
+            'cause' => $this->cause,
```

Comment:
I'm just working on an  API and I want to return errors to clients.
For example I have something like:

```php
function myAction(Request $request) {

  //...

  if ($form->isSubmitted() && $form->isValid()) {
    // Do something
  }  else {
    $data['errors'] = FormTool::getAllErrors($form);
  }

  return new JsonResponse($data);

}
```
In my FormTool:

```php
public static function getAllErrors(FormObject $form) {

  //...

  $errors = array_map(function(FormError $error) {

    if ($error instanceof JsonSerializable) {
      return $error;
    }

    return [
      'message' => $error->getMessage(),
      'messageTemplate' => $error->getMessageTemplate(),
      'messageParameters' => $error->getMessageParameters(),
      'messagePluralization' => $error->getMessagePluralization(),
      'cause' => $error->getCause()->__toString(), // not really useful
      ];

    }, $errors);

    return $errors;

}

```

I think that it could be useful if the FormError gives itself its own json serialisation.